server/src/main/java/org/elasticsearch/search/aggregations/bucket/adjacency/AdjacencyMatrixAggregationBuilder.java

```diff
@@ -204,17 +203,15 @@ protected AdjacencyMatrixAggregationBuilder doRewrite(QueryRewriteContext queryR
     @Override
     protected AggregatorFactory doBuild(AggregationContext context, AggregatorFactory parent, Builder subFactoriesBuilder)
         throws IOException {
-        int maxFilters = BooleanQuery.getMaxClauseCount();
+        int maxFilters = IndexSearcher.getMaxClauseCount();
         if (filters.size() > maxFilters) {
             throw new IllegalArgumentException(
                 "Number of filters is too large, must be less than or equal to: ["
                     + maxFilters
                     + "] but was ["
                     + filters.size()
-                    + "]."
-                    + "This limit can be set by changing the ["
-                    + SearchModule.INDICES_MAX_CLAUSE_COUNT_SETTING.getKey()
-                    + "] setting."
+                    + "].  "
+                    + "You can increase this limit by scaling up your java heap or number of CPUs"
```

Comment:
scaling up the number of CPUs would increase the size of the search threadpool and reduce the number of allowed clauses, so maybe just mention memory?